### PR TITLE
feat(infra): move to karnot self hosted runner

### PR DESCRIPTION
**Pull request by apoorvsadana**
Just testing out GH runner performance

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -198,7 +198,7 @@
 - feat(bonsai): Bumped bonsai lib to latest opti
 - refactor(generic): reduced runtime dependence on generics
 - fix(sync): Cleaned mc-sync isolating fetch process + added shared SyncStatus
-- feat(self-hosted): host our own runner
+- feat(ubuntu-latest): host our own runner
```
Comment (cchudant) [inline CHANGELOG.md]: lol :sweat_smile:

Comment (apoorvsadana) [reply CHANGELOG.md]: ah my bad, will undo this 😅

**Comment by k4kratik** [comment]
we are using "**buildjet**" as cache-provider here.
* 20GB per week.
* supports the current rust cache action - Swatinem/rust-cache@v2
* I feel download/upload of cache works with decent speed but compressing/decompressing takes a significant amount of time.

**Review by antiyro — APPROVED** [review]
lgtm I'll force merge this one to unlock the right context in further debugging of the CI

**Pull request merged**